apps/intellij/src/main/kotlin/dev/nx/console/generate/NxGeneratorSearchEvrywhereContributor.kt

```diff
@@ -0,0 +1,121 @@
+}
+
+class NxGeneratorSEContributor(val event: AnActionEvent) :
+    SearchEverywhereContributor<NxGenerator> {
+
+    private val project = event.getRequiredData(CommonDataKeys.PROJECT)
+
+    private val generators = runBlocking { project.service<NxlsService>().generators() }
+
+    private val generatorToOptionsCache: ConcurrentMap<NxGenerator, List<NxGeneratorOption>> =
+        ConcurrentHashMap()
+
+    override fun getSearchProviderId(): String = javaClass.name
+
+    override fun getGroupName(): String = "Nx"
+
+    override fun getSortWeight(): Int = 0
+
+    override fun showInFindResults(): Boolean = true
+
+    override fun getElementsRenderer(): ListCellRenderer<in NxGenerator> =
+        NxGeneratorListCellRenderer(
+            myIcon = AllIcons.Actions.GeneratedFolder,
+            alternatingRowColors = false,
+        )
+
+    override fun getDataForItem(element: NxGenerator, dataId: String): Any? {
+        return null
+    }
+
+    override fun processSelectedItem(
+        selected: NxGenerator,
+        modifiers: Int,
+        searchText: String
+    ): Boolean {
+        val service = project.service<NxlsService>()
+        val generatorOptions: List<NxGeneratorOption>? =
+            generatorToOptionsCache.computeIfAbsent(selected) {
+                runBlocking {
+                    service.generatorOptions(
+                        NxGeneratorOptionsRequestOptions(it.data.collection, it.name, it.path)
+                    )
+                }
+            }
+
+        val path =
+            if (ActionPlaces.isPopupPlace(event.place)) {
+                event.dataContext.getData(CommonDataKeys.VIRTUAL_FILE)?.path
+            } else {
+                null
+            }
+
+        val generatorWithOptions = NxGenerator(selected, generatorOptions)
+
+        val generatorContext = runBlocking {
+            path?.let { service.generatorContextFromPath(generatorWithOptions, path) }
+        }
+
+        val virtualFile = DefaultNxGenerateUiFile("Generate", project)
+        val fileEditorManager = FileEditorManager.getInstance(project)
+        if (fileEditorManager.isFileOpen(virtualFile)) {
+            fileEditorManager.closeFile(virtualFile)
+        }
+
+        fileEditorManager.openFile(virtualFile, true)
+
+        virtualFile.setupGeneratorForm(
+            NxGenerator(generator = generatorWithOptions, contextValues = generatorContext)
+        )
+        return true
+    }
+
+    override fun fetchElements(
+        pattern: String,
+        progressIndicator: ProgressIndicator,
+        consumer: Processor<in NxGenerator>
+    ) {
+
+        if (StringUtil.isEmptyOrSpaces(pattern)) {
+            return
+        }
+
+        val matcher = NameUtil.buildMatcher(pattern).build()
+
+        ProgressManager.getInstance()
+            .executeProcessUnderProgress(
+                { generators.filter { matcher.matches(it.name) }.forEach { consumer.process(it) } },
```

Comment:
how about allowing input like `nx g @nrwl/xxx` or something? so it's more like what you would type in the CLI?

Reply:
We should allow searching for generator like any symbols(a class, a table), and for this item I can for example execute an action, Open in UI or On Run Anything... 
That was the idea